nixos/modules/services/web-servers/lighttpd/nextcloud.nix

```diff
@@ -0,0 +1,118 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.lighttpd.nextcloud;
+  phpfpmSocketName = "/run/phpfpm/nextcloud.sock";
+in
+{
+  options.services.lighttpd.nextcloud = {
+
+    enable = mkEnableOption "Nextcloud in lighttpd";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.nextcloud;
+      defaultText = "pkgs.nextcloud";
+      description = "Nextcloud package to use.";
+    };
+
+    installPrefix = mkOption {
+      type = types.path;
+      default = "/var/lib/nextcloud";
+      description = ''
+        Where to install Nextcloud. By default, user files will be placed in
+        the data/ directory below the <option>installPrefix</option> directory.
+      '';
+    };
+
+    urlPrefix = mkOption {
+      type = types.str;
+      default = "/nextcloud";
+      example = "/";
+      description = ''
+        The prefix below the web server root URL to serve Nextcloud.
+      '';
+    };
+
+    vhostsPattern = mkOption {
+      type = types.str;
+      default = ".*";
+      example = "(myserver1.example|myserver2.example)";
+      description = ''
+        A virtual host regexp filter. Change it if you want Nextcloud to only
+        be served from some host names, instead of all.
+      '';
+    };
+
+  };
+
+  config = mkIf (config.services.lighttpd.enable && cfg.enable) {
+
+    systemd.services.lighttpd.preStart =
+      ''
+        echo "Setting up Nextcloud in ${cfg.installPrefix}/"
+        ${pkgs.rsync}/bin/rsync -a "${cfg.package}/" "${cfg.installPrefix}/"
```

Comment:
I had problems with one upgrade and I think Nextcloud suggested to use ./occ in case the upgrade took so long that the webUI timed out (or in case of error). I don't remember the details :-/